api/server.go

```diff
@@ -0,0 +1,30 @@
+package server
+
+import (
+	"database/sql"
+	"log"
+	"net/http"
+
+	"github.com/darkjedidj/cinema-service/api/halls"
+	hall "github.com/darkjedidj/cinema-service/internal/repository/halls"
+	"github.com/gorilla/mux"
+)
+
+type App struct {
+	Router *mux.Router
+}
+
+// New creates router with handler
+func (a *App) New(db *sql.DB) {
+
+	myRouter := mux.NewRouter().StrictSlash(false)
+	handler := halls.Handler{Repo: &hall.Repository{DB: db}}
+	myRouter.HandleFunc("/v1/halls/{id}", handler.HandleID)
```

Comment:
I know that gorilla/mux has something called Subrouters, i am not sure that it can help here, but check plz, it might help

Reply:
Routes can be used as subrouters: nested routes are only tested if the
  parent route matches. This is useful to define groups of routes that
  share common conditions like a host, a path prefix or other repeated
  attributes. As a bonus, this optimizes request matching. I guess its usefull, but wont solve problem with 2 Handle methods